src/concepts/airflow_gotchas.md

```diff
@@ -0,0 +1,72 @@
+
+`ExternalTaskMarker`s are generally wrapped into a `TaskGroup` and defined like:
+
+```python
+with TaskGroup('copy_deduplicate_all_external') as copy_deduplicate_all_external:
+    ExternalTaskMarker(
+        task_id="bhr_collection__wait_for_bhr_ping",    # name of task marker task
+        external_dag_id="bhr_collection",               # external downstream DAG
+        external_task_id="wait_for_bhr_ping",           # external downstream task ID
+        execution_date="{{ execution_date.replace(hour=5, minute=0).isoformat() }}",    # execution date calculated based on time differences in task schedules
+    )
+```
+
+Upstream dependencies are automatically determined between generated DAGs in bigquery-etl. If there are dependencies between DAGs in telemetry-airflow and bigquery-etl, then these dependencies need to be either added manually to the DAG definition or to the scheduling metadata of the scheduled query.
+
+## The DAG schedules are selected based on schedules of upstream dependencies
+
+The `schedule_interval` of a DAG should be set to a time that ensures that all upstream dependencies have likely finished before tasks in the DAG get executed. Airflow will send an email notification every time a task needs to be rescheduled due to upstream dependencies not having finished. To reduce the amount of notifications and avoid delays due to rescheduled tasks, the `schedule_interval` should be set based on when upstream tasks have finished.
+
+## Airflow triage
+
+To detect broken or stuck tasks, we set up an [Airflow triage process](https://mana.mozilla.org/wiki/display/DATA/Airflow+Triage+Process) that notifies tasks owners of problems with their Airflow tasks. Generally, DAGs are checked for failures or stuck tasks on a daily basis and problems are reported on [Bugzilla](https://bugzilla.mozilla.org/buglist.cgi?query_format=advanced&bug_status=UNCONFIRMED&bug_status=NEW&bug_status=ASSIGNED&bug_status=REOPENED&bug_status=RESOLVED&bug_status=VERIFIED&bug_status=CLOSED&status_whiteboard=%5Bairflow-triage%5D%20&classification=Client%20Software&classification=Developer%20Infrastructure&classification=Components&classification=Server%20Software&classification=Other&resolution=---&resolution=FIXED&resolution=INVALID&resolution=WONTFIX&resolution=INACTIVE&resolution=DUPLICATE&resolution=WORKSFORME&resolution=INCOMPLETE&resolution=SUPPORT&resolution=EXPIRED&resolution=MOVED&status_whiteboard_type=allwordssubstr&list_id=16121716).
+
+## Testing Airflow DAGs
```

Comment:
Possible addition - a screenshot might help too:
In case of a failure and after merging the solution to the problem, clear the logs for the failing task to allow the DAG to run again.

Reply:
Added